FirebaseAuth/Sources/Swift/User/UserInfoImpl.swift

```diff
@@ -18,6 +18,7 @@ import Foundation
 extension UserInfoImpl: NSSecureCoding {}
 
 @available(iOS 13, tvOS 13, macOS 10.15, macCatalyst 13, watchOS 7, *)
+@objc(FIRUserInfoImpl) // objc Needed for decoding old versions
```

Comment:
This fixed the `10.x → 11.x` behavior, but this fix will cause the same decoding failure for users created on `11.x`, so `11.0 ≤ 11.x → 11.6`.

Reply:
Good find!

How about only decoding FIRUserInfoImpl, as well as UserInfoImpl, and always encoding only UserInfoImpl?

Reply:
> this fix will cause the same decoding failure for users created on `11.x`, so `11.0 ≤ 11.x → 11.6`.

this actually isn't the case. this path also succeeds with the fix. Adding `NSClassFromString("FIRUserInfoImpl").self` to the list of decoding classes actually evaluates to "FirebaseAuth.UserInfoImpl", so:

let classes = [UserInfoImpl.self, NSClassFromString("FIRUserInfoImpl")!.self]
print(classes) // ["FirebaseAuth.UserInfoImpl", "FirebaseAuth.UserInfoImpl"]

Reply:
To test, I wiped simulator, installed 11.5.0, signed in with GSI provider. Then quit, and installed this fix. Provider data was successfully decoded so this fix appears to be forward compatible.